### PR TITLE
Allow modules to alter the default field storage value through a hook.

### PR DESCRIPTION
**Pull request by svendecabooter**

### CODE REVIEW

File: src/FieldEncryptProcessEntities.php [hunk 1]
```diff
@@ -279,7 +279,14 @@ protected function processValue(ContentEntityInterface $entity, FieldItemListInt
       // Return value to store for unencrypted property.
       // We can't set this to NULL, because then the field values are not saved,
       // so we can't replace them with their unencrypted value on load.
-      return '[ENCRYPTED]';
+      $unencrypted_storage_value = '[ENCRYPTED]';
```
Comment (nerdstein) [inline src/FieldEncryptProcessEntities.php:282]: What about numeric values?

**Comment by nerdstein** [comment]
This looks great as well, thanks. Left one small comment. Maybe we have default values by field type, e.g. a date, a string, a number...

**Comment by nerdstein** [comment]
Please feel free to merge and/or make separate issues where relevant

**Comment by svendecabooter** [comment]
Settings default values for some less obvious field types could be considered, but we have to be careful of not skewing the actual data. For example you'd have to pick a date that hopefully can't be confused with real date values.

I'd have to think about whether it would be better to explicitly force site builders to consider this scenario (by having to implement the hook, and hopefully understanding the consequences), versus doing that for them...

**Comment by nerdstein** [comment]
Do you want to merge this in for now and explore this use case further in another issue?

**Comment by svendecabooter** [comment]
Yeah sounds good.

**Comment by nerdstein** [comment]
Merged - please create the issue so we can track it

**Pull request merged**